### ISSUE TITLE
[Fungible Asset][Framework] FA APT lacks testing APIs, prohibiting effective development

### ISSUE THREAD

**Issue by alnoki** [aptos-labs/aptos-core#11839]
Per offline discussion with @lightmark @davidiw @movekevin 

The Coin -> Fungible Asset (FA) transition at #11224 implies that new Move packages should be developed around the concept of FA APT.

However, presently there is no `#[test_only]` function in the Aptos Framework for ensuring that the required `Object<Metadata>` for FA APT exists under `0x1`, hence developers cannot reference FA-style APT in their Move code. Moreover, the mere definition of FA APT is insufficient since users will also need to be able to acquire FA APT in a quasi-faucet style during testing. Hence:

- [x] Add `#[test_only]` function of the form `register_fungible_asset_apt_for_testing()` that creates `Object<Metadata>` for APT under `0x1` as appropriate
- [x] #11974

**Comment by alnoki** [comment]
@lightmark I see #11974 is just a blank issue, but I think it may have missed the point

The idea is that during test runtime developers can get FA APT, hence the second checkmark I put above was a sort of implementation possibility (e.g. return the `MintRef` for FA APT)

Mainly the functionality is to ensure two things:

1. The `Metadata` for FA APT exists during test runtime
2. Devs can mint FA APT during test runtime, like a faucet (burn and forced transfer functions would help too)

Something like

```move
#[test_only]
public fun ensure_fa_apt_metadata_at_0x1_for_testing()

#[test_only]
public fun mint_fa_apt_to_users_primary_fungible_store(user: address)

#[test_only]
public fun mint_fa_apt(amount: u64): FungibleAsset
```

**Comment by lightmark** [comment]
basically you can call `initialize_for_test` in aptos_coin.move with 0x1 and then mint to any address with 0x1 signer with `mint`. Then you could fetch APT coin and convert it to FA. I think that's the only way to do it if you really want fake APT FA.
I don't think framework should provide such helper functions in the framework because as a developer the test should not depend on a specific type. I will test 0x1 and AptosCoin is matched in aptos_coin.move

**Comment by alnoki** [comment]
@lightmark thank you for the references

Accordingly I've created #12028

**Comment by alnoki** [comment]
Closed per #12028
